### ISSUE TITLE
P/Invoked (cross platform) native library isn't working on OS X (10.14.2)

### ISSUE THREAD

**Issue by define-private-public** [dotnet/core#4362]
About a year ago I built this package called [Bassoon](https://www.nuget.org/packages/Bassoon), which offers cross platform (Windows, Linux, and OS X) audio playback and decoding.  Under the hood, it uses `portaudio` (playback) and `libsndfile` (file format decoding).  This all was working fine a couple of months ago.  Recently I' wanted to switch the build system to where the user has to use `vcpkg` to build those native dependencies.  So far this has been good on Linux and Windows.  But OS X has been a bit of a problem.

I run off of a slightly older Mac Mini (10.14.2), but I still was able to install `vcpkg` and have it build those native libraries.  Having .NET Core (runtimes 2.1 and 3.1) do a P/Invoke call to `portaudio` is fine, but `libsndfile` was causing the problems.  It's just failing to load it, where it was working before.  In the Bassoon repo, there is a sample project called `FileInfo` I tried to run.  As soon as it tries to call `libsndfile.Version()` the program crashes saying that it wasn't able to find the `libsndfile`^ dylib.

The first error message said to try turning on the `DYLD_PRINT_LIBRARIES` variable.  It showned that at first `libsndfile` was being loaded, but then unloaded immediately:

    dyld: loaded: /Users/ben/Desktop/Bassoon/third_party/lib//libsndfile.dylib
    dyld: unloaded: /Users/ben/Desktop/Bassoon/third_party/lib//libsndfile.dylib

Any ideas as to why this is happening?  This kind really sucks since it was working before, but now isn't.  And I'm not that proficient with OS X debugging.

My end goal is create a NuGet package which has all of these native deps in there, so any users don't need to go through this whole "build native libs and set environment variables" madness.

---
^ Note: `vcpkg` builds a file called `libsndfile-shared.dylib`.  I removed that `-shared` part because on windows the built DLL's name is a little different too.  It shouldn't affect the "not able to find" part though.

**Comment by scalablecory** [comment]
Can you please shrink this down to a small repro? It isn't obvious here what might be the problem.

**Comment by define-private-public** [comment]
If you clone the Bassoon repo, then follow the [Developing](https://gitlab.com/define-private-public/Bassoon#id6) section of the README to get the native libraries setup, you should be able to follow along.

**Comment by define-private-public** [comment]
I figured out the issue.  It's not anything related to .NET core (so I'll close this ticket).  It was happening because on OS X I renamed the library from `libsndfile-shared.dylib` to `libsndfile.dylib`.  For anyone who's interested in the details:

---

I ran `otool -L libsndfile.dylib`.  It reported this:

    $ otool -L libsndfile.dylib 
    libsndfile.dylib:
            @rpath/libsndfile-shared.1.dylib (compatibility version 1.0.0, current version 1.0.29)
            ...

So it's still kinda looking for the `-shared` part.  I've worked with this issue before, so I think I can handle it from here.
